modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -161,6 +118,8 @@ public TimeValue getCompareAndExchangeTimeToLive() {
     // issue
     private abstract static class IgnoreNoResponseMetricsCollector extends RequestMetricCollector {
```

Comment:
Do we still need this abstract class? It is only inherited once, should it be in-lined with the only usage?

Reply:
Unfortunately yes. The `IgnoreNoResponseMetricsCollector` type allows us to access its `counter` field since the super class `RequestMetricCollector` is owned by AWS.

Reply:
If we would like to have our own interface, should we convert it to a regular class (that taces care of incrementing the counter from the no-null request) and use it in our code?

Reply:
That's fair. I pushed 2301f27543e13f549a31ba7be8118e046a682f72